src/ext/NetFx/ca/netfxca.cpp

```diff
@@ -879,3 +907,132 @@ extern "C" UINT __stdcall ExecNetFx(
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        ExitOnFailure(hr, "failed to get temp path value");
+    }
+
+    // Open view on .NET compatibility check table
+    hr = WcaOpenExecuteView(vscDotNetCompatibilityCheckQuery, &hView);
+    ExitOnFailure(hr, "failed to open view on Wix4NetFxDotNetCompatibilityCheck table");
+
+    // Go through all records and run NetCorCheck.exe for each
+    while (S_OK == (hr = WcaFetchRecord(hView, &hRec)))
+    {
+        HANDLE hProcess = INVALID_HANDLE_VALUE;
+        DWORD dwExitCode = 0;
+
+        // Extract NetCoreCheck.exe for platform to temp directory
+        hr = WcaGetRecordString(hRec, platform, &pwzPlatform);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.Platform");
+
+        hr = GetNetCoreCheckBinaryId(pwzPlatform, &pwzNetCoreCheckBinaryId);
+        ExitOnFailure(hr, "failed to get NetCoreCheck binary id for platform %ls", pwzPlatform);
+
+        hr = StrAllocFormatted(&pwzNetCoreCheckFilePath, L"%ls%ls", pwzTempPath, L"NetCoreCheck.exe");
+        ExitOnFailure(hr, "failed to set NetCoreCheck file path");
+
+        hr = WcaExtractBinaryToFile(pwzNetCoreCheckBinaryId, pwzNetCoreCheckFilePath);
+        ExitOnFailure(hr, "failed to extract NetCoreCheck from binary '%ls' to file %ls", pwzNetCoreCheckBinaryId, pwzNetCoreCheckFilePath);
+
+        // Read all NetCoreCheck.exe parameters and variable
+        hr = WcaGetRecordString(hRec, runtimeType, &pwzRuntimeType);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.RuntimeType");
+
+        hr = WcaGetRecordString(hRec, version, &pwzVersion);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.Version");
+
+        hr = WcaGetRecordString(hRec, rollForward, &pwzRollForward);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.RollForward");
+
+        hr = WcaGetRecordString(hRec, variable, &pwzVariable);
+        ExitOnFailure(hr, "failed to get Wix4DotNetCompatibilityCheck.Variable");
+
+        // Run NetCoreCheck.exe and store its result in variable
+        hr = StrAllocFormatted(&pwzCommandLine, L"-n %ls -v %ls -r %ls", pwzRuntimeType, pwzVersion, pwzRollForward);
+        ExitOnFailure(hr, "failed to set NetCoreCheck command line");
+        WcaLog(LOGMSG_VERBOSE, "Command: %ls %ls", pwzNetCoreCheckFilePath, pwzCommandLine);
+
+        hr = ProcExec(pwzNetCoreCheckFilePath, pwzCommandLine, SW_HIDE, &hProcess);
+        ExitOnFailure(hr, "failed to run NetCoreCheck from binary '%ls' with command line: %ls %ls", pwzNetCoreCheckBinaryId, pwzNetCoreCheckFilePath, pwzCommandLine);
+
+        hr = ProcWaitForCompletion(hProcess, INFINITE, &dwExitCode);
+        ExitOnFailure(hr, "failed to finish NetCoreCheck from binary '%ls' with command line: %ls %ls", pwzNetCoreCheckBinaryId, pwzNetCoreCheckFilePath, pwzCommandLine);
+        WcaLog(LOGMSG_VERBOSE, "Exit code: %lu", dwExitCode);
+        CloseHandle(hProcess);
```

Comment:
Normally we would use the `ReleaseHandle` macro. Also, the handle needs to be closed after `LExit` to make sure it isn't leaked if there is an error.